src/test/java/io/strimzi/kafka/bridge/http/HttpCorsTests.java

```diff
@@ -0,0 +1,193 @@
+                    .request(HttpMethod.OPTIONS, 8080, "localhost", "/ready")
+                    .putHeader("Origin", "https://strimzi.io")
+                    .putHeader("Access-Control-Request-Method", "GET")
+                    .send(ar -> {
+                        context.verify(() -> {
+                            assertThat(ar.result().statusCode(), is(200));
+                            assertThat(ar.result().getHeader("access-control-allow-origin"), is(origin));
+                            assertThat(ar.result().getHeader("access-control-allow-headers"), is("Access-Control-Allow-Origin,origin,x-requested-with,Content-Type,accept"));
+                            context.completeNow();
+                        });
+
+                    })));
+
+        } else {
+            context.completeNow();
+        }
+    }
+
+    @Test
+    public void testMethodNotAllowedRequest(VertxTestContext context) {
+        createWebClient(true, "POST,PUT,DELETE,OPTIONS,PATCH", context);
+        final String origin = "https://strimzi.io";
+
+        if (!"TRUE".equalsIgnoreCase(System.getenv("STRIMZI_USE_SYSTEM_BRIDGE"))) {
+            vertx.deployVerticle(httpBridge, context.succeeding(id -> client
+                    .request(HttpMethod.OPTIONS, 8080, "localhost", "/ready")
+                    .putHeader("Origin", "https://strimzi.io")
+                    .putHeader("Access-Control-Request-Method", "GET")
+                    .send(ar -> {
+                        context.verify(() -> {
+                            assertThat(ar.result().statusCode(), is(200));
+                            assertThat(ar.result().getHeader("access-control-allow-origin"), is(origin));
+                            assertThat(ar.result().getHeader("access-control-allow-headers"), is("Access-Control-Allow-Origin,origin,x-requested-with,Content-Type,accept"));
+                            List<String> list = Arrays.asList(ar.result().getHeader("access-control-allow-methods").split(","));
+                            assertThat(list, not(hasItem("GET")));
+                            context.completeNow();
+                        });
+
+                    })));
+
+        } else {
+            context.completeNow();
+        }
+    }
+
+    private void createWebClient(boolean corsEnabled, String methodsAllowed, VertxTestContext context) {
+        config.put(HttpConfig.HTTP_CORS_ENABLED, String.valueOf(corsEnabled));
+        config.put(HttpConfig.HTTP_CORS_ALLOWED_ORIGINS, "https://strimzi.io");
+        config.put(HttpConfig.HTTP_CORS_ALLOWED_METHODS, methodsAllowed != null ? methodsAllowed : "GET,POST,PUT,DELETE,OPTIONS,PATCH");
+
+        if (!"TRUE".equalsIgnoreCase(System.getenv("STRIMZI_USE_SYSTEM_BRIDGE"))) {
+            bridgeConfig = BridgeConfig.fromMap(config);
+            httpBridge = new HttpBridge(bridgeConfig);
+            httpBridge.setHealthChecker(new HealthChecker());
+        }
```

Comment:
I think we should have a different method for the bridge configuration because it's not part of creating a web client (as the `createWebClient` drives to think).

Reply:
Yes I will move this logic into
configureBridge(); private method